csrc/scheduler/normalization_inner_outer.h

```diff
@@ -26,11 +26,10 @@ class HeuristicDataCache;
 
 class InnerOuterPersistentKernelScheduler : public SchedulerEntry {
  public:
-  // This scheduler has very high register pressure due to extra registers to
-  // store intermediate outer reduction results. So prefer to allow 255
-  // registers per thread and then the max threads per block is 256.
   constexpr static int64_t threads_per_block_min = 128l;
-  constexpr static int64_t threads_per_block_max = 256l;
+  // 512 threads may lead to higher occupancy than 256 threads,
+  // e.g. when shared memory only allows 1 block per SM.
```

Comment:
deleted.
For example, each SM has 200KB avilable smem, one normalization requires 200KB shared memory, regardless of threads per block, each SM can only process one normalization, which is one block, since we only do block reduction. So under this condition, using a larger threads per block leads to higher occupancy.

Reply:
logging offline conversation.

Because we only use block reduction in the persistent kernel, each CTA needs to hold the entire row of data in shared memory, which could cap the occupancy to be 1 block per sm. i.e. larger block size would give us higher overall occupancy.